pkg/cmd/cli/cmd/env.go

```diff
@@ -268,14 +285,14 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 		}
 	}
 	if one && skipped == len(infos) {
-		return fmt.Errorf("the %s %s is not a pod or does not have a pod template", infos[0].Mapping.Resource, infos[0].Name)
+		return fmt.Errorf("%s/%s is not a pod or does not have a pod template", infos[0].Mapping.Resource, infos[0].Name)
 	}
 
 	if list {
 		return nil
 	}
 
-	objects, err := resource.AsVersionedObject(infos, false, outputVersion)
+	objects, err := resource.AsVersionedObjects(infos, outputVersion)
```

Comment:
--output-version still matters if they want to print the objects
I'm not positive how --api-version behaves if you specify a version the server doesn't speak. Test with v1beta3 and see if it fails or negotiates to v1